apex_launchtest/apex_launchtest/apex_runner.py

```diff
@@ -177,6 +196,10 @@ def _print_process_output_summary(self):
                 print("{}".format(io.text.decode('ascii')))
             print("#" * (len(process.process_name) + 21))
 
+    def _bind_test_context_to_tests(self, context, test_suite):
+        for k, v in context.items():
+            self._give_attribute_to_tests(v, k, test_suite)
```

Comment:
@pbaughman hmm, why not passing them as arguments for the tests instead? wouldn't that preclude attribute clashes?

Reply:
@hidmic I don't have a ton of control over what's passed into the tests, although maybe someone with a better understanding of unittest can point me in the right direction.  We delegate that to the unittest.TextTestRunner [here](https://github.com/ApexAI/apex_rostest/blob/3233d7992c48423af25781270650935b8efaf1f3/apex_launchtest/apex_launchtest/apex_runner.py#L161) and [here](https://github.com/ApexAI/apex_rostest/blob/3233d7992c48423af25781270650935b8efaf1f3/apex_launchtest/apex_launchtest/apex_runner.py#L129)

The [runner calls run](https://github.com/python/cpython/blob/dc078947a5033a048d804e244e847b5844734439/Lib/unittest/runner.py#L176) on the test suite which [calls run](https://github.com/python/cpython/blob/dc078947a5033a048d804e244e847b5844734439/Lib/unittest/suite.py#L122) on the test case which [executes the actual test method](https://github.com/python/cpython/blob/master/Lib/unittest/case.py#L642) and doesn't pass arguments

Reply:
Thinking about this on the bus last night. . .

Maybe I could pre-process the list of test methods and bind the arguments before I pass them into TextTestRunner.run

I think there's a functools method for that